src/lib/navigation/wizard-state.model.ts

```diff
@@ -50,13 +50,37 @@ export class WizardState {
     } else {
       return this._defaultStepIndex;
     }
-  };
+  }
+
+  /**
+   * The index of the currently visible and selected step inside the wizardSteps QueryList.
+   * If this wizard contains no steps, currentStepIndex is -1
+   */
+  private _currentStepIndex = -1;
+
+  /**
+   * The index of the currently visible and selected step inside the wizardSteps QueryList.
+   * If this wizard contains no steps, currentStepIndex is -1
+   */
+  public get currentStepIndex(): number {
+    return this._currentStepIndex;
+  }
 
   /**
    * The index of the currently visible and selected step inside the wizardSteps QueryList.
    * If this wizard contains no steps, currentStepIndex is -1
    */
-  public currentStepIndex = -1;
+  public set currentStepIndex(index: number) {
+    if (index === this._currentStepIndex) {
+      return;
+    }
+
+    this._currentStepIndex = index;
+
+    if (this._currentStepIndex >= 0) {
+      this.stepChanged.next(this.wizardSteps[this._currentStepIndex]);
```

Comment:
This should be moved to the navigation modes

Reply:
I thought of that at first but then for each type of navigation mode (strict and etc) need to be the same code, so why don’t we make it once in a service that have current index already